File: app/src/main/java/com/baro/ui/spalsh/SplashActivity.java
package com.baro.ui.spalsh;

import androidx.annotation.RequiresApi;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.AsyncTask;
import android.os.Build;
import android.os.Bundle;
import android.os.Handler;
import android.os.Parcelable;
import android.view.View;
import android.widget.ProgressBar;
import android.widget.Toast;

import com.baro.R;
import com.baro.constants.FileEnum;
import com.baro.constants.IntentEnum;
import com.baro.constants.JSONEnum;
import com.baro.helpers.FileHelper;
import com.baro.helpers.JSONHelper;
import com.baro.models.User;
import com.baro.ui.main.MainActivity;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.File;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.UUID;

public class  SplashActivity extends AppCompatActivity {

    private static int SPLASH_TIME_OUT = 1000;
    private ProgressBar progressBar;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        configureProgressBar();


        if (savedInstanceState == null) {

            new Handler().postDelayed((Runnable) () -> {
                getCurrentlyLoggingInUser();
            }, SPLASH_TIME_OUT);

        }
    }

    private void configureProgressBar() {
        progressBar = findViewById(R.id.progress_bar);
    }

    private void getCurrentlyLoggingInUser() {
        UserCredentialsTask userCredentialsTask = new UserCredentialsTask();
        userCredentialsTask.execute();
    }


    private class UserCredentialsTask extends AsyncTask<Void, Void, User> {

        @RequiresApi(api = Build.VERSION_CODES.O)
        @Override
        protected User doInBackground(Void... voids) {

            Path userMetaDataPath = Paths.get(
                    String.valueOf(getExternalFilesDir(null)),
                    FileEnum.USER_DIRECTORY.key,
                    FileEnum.META_DATA_FILE.key);

            File userMetaDataFile = FileHelper.getFileAtPath(userMetaDataPath);

            if (userMetaDataFile != null) {
                String contentMetaData = FileHelper.readFile(userMetaDataFile);
                JSONObject jsonMetaData = JSONHelper.createJSONFromString(contentMetaData);
                try {
                    User user = new User(
                            UUID.fromString((String) jsonMetaData.get(JSONEnum.USER_UUID_KEY.key)),
                            (String) jsonMetaData.get(JSONEnum.USER_NAME_KEY.key));
                    return user;
                } catch (JSONException e) {
                    return null;
                }
            }
            return null;

        }

        protected void onPostExecute(User result) {


            if (result != null) {
                Toast.makeText(SplashActivity.this,
                        getString(R.string.toast_logging_in_user) + result.getUsername(),
                        Toast.LENGTH_LONG)
                        .show();
                Intent startMainActivity = new Intent(
                        SplashActivity.this,
                        MainActivity.class);
                // Passing User to MainActivity
                startActivity(startMainActivity);
                SplashActivity.this.finish();
            } else {
                Toast.makeText(SplashActivity.this,
                        getString(R.string.toast_no_username_found),
                        Toast.LENGTH_LONG)
                        .show();

                progressBar.setVisibility(View.INVISIBLE);

                getSupportFragmentManager().beginTransaction()
                        .setReorderingAllowed(true)
                        .add(R.id.fragment_container_view, SplashLoggingFragment.class, null)
                        .commit();

            }
        }
    }
}

File: app/src/main/java/com/baro/ui/splash/SplashActivity.kt
package com.baro.ui.splash


import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager
import android.net.NetworkCapabilities
import android.os.Build
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.util.Log
import android.view.View
import android.widget.ProgressBar
import android.widget.Toast
import androidx.annotation.RequiresApi
import androidx.appcompat.app.AppCompatActivity

import com.baro.R
import com.baro.constants.AppTags
import com.baro.helpers.*
import com.baro.models.User
import com.baro.ui.main.MainActivity
import com.google.firebase.auth.FirebaseAuth
import kotlinx.coroutines.launch
import kotlinx.coroutines.runBlocking

class SplashActivity : AppCompatActivity() {

    private var progressBar: ProgressBar? = null

    @RequiresApi(Build.VERSION_CODES.O)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        configureProgressBar()
        if (savedInstanceState == null) {
            Handler(Looper.getMainLooper()).postDelayed({ getCurrentlyLoggingInUser() }, SPLASH_TIME_OUT.toLong())
        }
    }


    private fun configureProgressBar() {
        progressBar = findViewById(R.id.progress_bar)
    }

    @RequiresApi(Build.VERSION_CODES.O)
    private fun getCurrentlyLoggingInUser() {

        runBlocking {
            launch {
                var user = AsyncHelpers().verifyUserCredentials(getExternalFilesDir(null))
                onUserLoginCheckDone(user)
            }
        }

    }

    companion object {
        const val SPLASH_TIME_OUT = 1000
    }


    @RequiresApi(Build.VERSION_CODES.M)
    private fun onUserLoginCheckDone(result: User?) {
        if (result != null) {

            if (isOnline(this.applicationContext)) {
                if (FirebaseAuth.getInstance().currentUser?.uid == null) {
                    supportFragmentManager.beginTransaction()
                        .setReorderingAllowed(true)
                        .add(R.id.fragment_container_peer_connection,
                            SplashLoggingFirebaseFragment::class.java, null)
                        .commit()
                }
            }
            val startMainActivity = Intent(
                this,
                MainActivity::class.java)

            startActivity(startMainActivity)
            finish()
        } else {
            Toast.makeText(this,
                    getString(R.string.toast_no_username_found),
                    Toast.LENGTH_LONG)
                    .show()

            progressBar!!.visibility = View.GONE


            supportFragmentManager.beginTransaction()
                    .setReorderingAllowed(true)
                    .add(R.id.fragment_container_peer_connection, SplashLoggingFragment::class.java, null)
                    .commit()
        }
    }

    @RequiresApi(Build.VERSION_CODES.M)
    fun isOnline(context: Context?): Boolean {
        val connectivityManager =
            context?.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        if (connectivityManager != null) {
            val capabilities =
                connectivityManager.getNetworkCapabilities(connectivityManager.activeNetwork)
            if (capabilities != null) {
                when {
                    capabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR) -> {
                        Log.i("Internet", "NetworkCapabilities.TRANSPORT_CELLULAR")
                        return true
                    }
                    capabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) -> {
                        Log.i("Internet", "NetworkCapabilities.TRANSPORT_WIFI")
                        return true
                    }
                    capabilities.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET) -> {
                        Log.i("Internet", "NetworkCapabilities.TRANSPORT_ETHERNET")
                        return true
                    }
                }
            }
        }
        return false
    }
}
